staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -623,10 +653,63 @@ func preprocessDeletionListForMerging(key string, original map[string]interface{
 	return false, false, "", nil
 }
 
+// processReplaceKeys processes replaceKeys list.
+// When MergeParallelList is true, we validate the patch and
+// then clear the fields that are not in the list.
+func processReplaceKeys(original, patch map[string]interface{}, options MergeOptions) error {
+	replaceKeysInPatch, foundInPatch := patch[replaceKeysDirective]
+	if !foundInPatch {
+		return nil
+	}
+	delete(patch, replaceKeysDirective)
+
+	if !options.MergeParallelList {
```

Comment:
// If original is actually a patch, make sure the retainkeys directives are the same in both patches.